lock/src/main/java/org/infinispan/lock/impl/lock/ClusteredLockImpl.java

```diff
@@ -290,23 +390,48 @@ private void tryLock(TryLockRequestHolder requestHolder) {
       }
       if (requestHolder.hasTimeout()) pendingRequests.offer(requestHolder);
 
-      readWriteMap.eval(lockKey, new LockFunction(requestHolder.requestId, requestHolder.requestor)).whenComplete((lockResult, ex) -> {
-         requestHolder.handleLockResult(lockResult, ex);
-      });
+      if (isChangeViewUnlockInProgress()) {
+         requestHolder.handleLockResult(false, null);
+      } else {
+         readWriteMap.eval(lockKey, new LockFunction(requestHolder.requestId, requestHolder.requestor)).whenComplete((lockResult, ex) -> {
+            requestHolder.handleLockResult(lockResult, ex);
+         });
+      }
    }
 
    @Override
    public CompletableFuture<Void> unlock() {
       log.tracef("unlock called from %s", originator);
       CompletableFuture<Void> unlockRequest = new CompletableFuture<>();
-      readWriteMap.eval(lockKey, new UnlockFunction(originator)).whenComplete((lockResult, ex) -> {
+
+      if (isChangeViewUnlockInProgress()) {
+         pendingUnlocks.offer(new UnlockRequestHolder(originator, unlockRequest));
```

Comment:
What happens if `forceUnlockForLeavingMembers` decrements `viewChangeUnlockHappening` *and* calls `retryPendingUnlocks()` between lines 407 and 408?

Reply:
retryPendingUnlocks removed